src/java/org/apache/cassandra/db/AbstractBufferClusteringPrefix.java

```diff
@@ -42,10 +42,18 @@ public ByteBuffer[] getBufferArray()
         return getRawValues();
     }
 
-    public ClusteringPrefix<ByteBuffer> minimize()
+    public ClusteringPrefix<ByteBuffer> retainable()
     {
         if (!ByteBufferUtil.canMinimize(values))
             return this;
-        return new BufferClustering(ByteBufferUtil.minimizeBuffers(values));
+
+        ByteBuffer[] values = ByteBufferUtil.minimizeBuffers(this.values);
```

Comment:
Maybe we could use a new name:
```suggestion
        ByteBuffer[] minimizedValues = ByteBufferUtil.minimizeBuffers(values);
```

Reply:
Done